one_fm/hiring/doctype/onboard_employee/onboard_employee.js

```diff
@@ -305,6 +306,24 @@ var btn_inform_applicant = function(frm) {
 	}).addClass('btn-primary');
 };
 
+var check_signature = function(frm) {
+	if(frm.doc.declaration_of_electronic_signature){
+		frappe.call({
+			doc: frm.doc,
+			method: 'check_signature_status',
+			callback: function(r) {
+				console.log(r.message)
+				if(!r.exc){
+					return true
+				}
+			},
+			freeze: true,
+			freeze_message: __('reloading ...!')
+		});
+		refresh_field("declaration_status");
+	}
+}
+
```

Comment:
Instead of updating the status from Onboard Employee doctype, you can make use of the method in Electronic Signature Declaration `update_onboarding_doc` method

Reply:
I tried that. But it didn't update anything.

Reply:
`def update_onboarding_doc(self, cancel=False):
		if self.onboard_employee:
			onboard_employee = frappe.get_doc('Onboard Employee', self.onboard_employee)
			if cancel:
				onboard_employee.declaration_of_electronic_signature = ''
			else:
				onboard_employee.declaration_of_electronic_signature = self.name
                                 if self.applicant_signature:
                                          onboard_employee.electronic_signature_status = True

			onboard_employee.save(ignore_permissions=True)`